From: Stephen Rothwell @ 2020-07-08 10:46 UTC (permalink / raw)
To: Andrew Morton, Theodore Ts'o, Eric Biggers
Cc: Linux Next Mailing List, Linux Kernel Mailing List, Satya Tangirala, Waiman Long


[-- Attachment #1: Type: text/plain, Size: 1172 bytes --]

Hi all,

Today's linux-next merge of the akpm-current tree got a conflict in:

  fs/crypto/keysetup_v1.c

between commit:

  73f43e98d01e ("fscrypt: add inline encryption support")

from the fscrypt tree and commit:

  f15de703d326 ("mm, treewide: rename kzfree() to kfree_sensitive()")

from the akpm-current tree.

I fixed it up (see below) and can carry the fix as necessary. This
is now fixed as far as linux-next is concerned, but any non trivial
conflicts should be mentioned to your upstream maintainer when your tree
is submitted for merging.  You may also want to consider cooperating
with the maintainer of the conflicting tree to minimise any particularly
complex conflicts.

-- 
Cheers,
Stephen Rothwell

diff --cc fs/crypto/keysetup_v1.c
index a52686729a67,c8a930f8faf2..000000000000
--- a/fs/crypto/keysetup_v1.c
+++ b/fs/crypto/keysetup_v1.c
@@@ -154,8 -154,8 +154,8 @@@ struct fscrypt_direct_key 
  static void free_direct_key(struct fscrypt_direct_key *dk)
  {
  	if (dk) {
 -		crypto_free_skcipher(dk->dk_ctfm);
 +		fscrypt_destroy_prepared_key(&dk->dk_key);
- 		kzfree(dk);
+ 		kfree_sensitive(dk);
  	}
  }
  

[-- Attachment #2: OpenPGP digital signature --]
[-- Type: application/pgp-signature, Size: 488 bytes --]

